x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -215,21 +228,25 @@ export class SessionIndex {
       await this.indexInitialization;
     }
 
-    const { sid, ...sessionValueToStore } = sessionValue;
     try {
-      const { _primary_term: primaryTerm, _seq_no: sequenceNumber } =
-        await this.options.elasticsearchClient.create({
-          id: sid,
-          // We cannot control whether index is created automatically during this operation or not.
-          // But we can reduce probability of getting into a weird state when session is being created
-          // while session index is missing for some reason. This way we'll recreate index with a
-          // proper name and alias. But this will only work if we still have a proper index template.
-          index: this.indexName,
-          body: sessionValueToStore,
-          refresh: 'wait_for',
-        });
+      let { body, statusCode } = await this.writeNewSessionDocument(sessionValue, {
+        ignore404: true,
+      });
+
+      if (statusCode === 404) {
+        this.options.logger.warn(
+          'Attempted to create a new session, but session index or alias was missing.'
+        );
+        await this.ensureSessionIndexExists();
+        ({ body, statusCode } = await this.writeNewSessionDocument(sessionValue, {
```

Comment:
> Looks like you unintentionally returned unused `statusCode` back 🙂

This is actually required to avoid an eslint error, because both `body` and `statusCode` are part of the same `let` block above. If I remove `statusCode` here, it complains that `statusCode` isn't a const above. I could either fix this by reintroducing the unused `statusCode` reference here, _or_ break up the destructured assignment above into multiple lines. Reintroducing `statusCode` here felt less bad to me, but I don't feel strongly about this if you would prefer that I change it around.

Reply:
> This is actually required to avoid an eslint error, because both body and statusCode are part of the same let block above. If I remove statusCode here, it complains that statusCode isn't a const above.

Ahh, I see now, thanks for clarifying! Please, keep it as is then.